src/ConstructiveSolidGeometry/VolumePrimitives/RegularPrism.jl

```diff
@@ -0,0 +1,90 @@
+struct RegularPrism{N,T,TR,TZ} <: AbstractVolumePrimitive{T}
+    r::TR
+    z::TZ
+    
+    function RegularPrism(  N::Integer,
+                            ::Type{T},
+                             r::Union{T, <:AbstractInterval{T}},
+                             z::Union{T, <:AbstractInterval{T}}) where {T}
+        @assert typeof(N) <: Integer && N >= 3 "Failed constructing RegularPrism with $(N) points in the polygon base (needs at least 3 points)."
+        new{N, T,typeof(r),typeof(z)}(r,z)
+    end
+end
+
+
+# Convenience functions
+const TriangularPrism{T,TR,TZ} = RegularPrism{3,T,TR,TZ}
+const QuadraticPrism{T,TR,TZ}  = RegularPrism{4,T,TR,TZ}
+const PentagonalPrism{T,TR,TZ} = RegularPrism{5,T,TR,TZ}
+const HexagonalPrism{T,TR,TZ}  = RegularPrism{6,T,TR,TZ}
+
+TriangularPrism(args...) = RegularPrism(3, args...)
+QuadraticPrism(args...)  = RegularPrism(4, args...)
+PentagonalPrism(args...) = RegularPrism(5, args...)
+HexagonalPrism(args...)  = RegularPrism(6, args...)
+
+print(io::IO, rp::TriangularPrism{T, TR, TZ}) where {T,TR,TZ} = print(io, "TriangularPrism{$(T), $(TR), $(TZ)}($(rp.r), $(rp.z))")
+print(io::IO, rp::QuadraticPrism{T, TR, TZ})  where {T,TR,TZ} = print(io, "QuadraticPrism{$(T), $(TR), $(TZ)}($(rp.r), $(rp.z))")
+print(io::IO, rp::PentagonalPrism{T, TR, TZ}) where {T,TR,TZ} = print(io, "PentagonalPrism{$(T), $(TR), $(TZ)}($(rp.r), $(rp.z))")
+print(io::IO, rp::HexagonalPrism{T, TR, TZ})  where {T,TR,TZ} = print(io, "HexagonalPrism{$(T), $(TR), $(TZ)}($(rp.r), $(rp.z))")
+
+
+# Constructors
+function RegularPrism(N::Integer; rInner = 0, rOuter = 1, zMin = -1, zMax = 1)
+    T = float(promote_type(typeof.((rInner, rOuter, zMin, zMax))...))
+    r = rInner == 0 ? T(rOuter) : T(rInner)..T(rOuter)
+    z = zMax == -zMin ? T(zMax) : T(zMin)..T(zMax)
+    RegularPrism(N, T, r, z)
+end
+RegularPrism(N::Integer, rInner, rOuter, zMin, zMax) = RegularPrism(N; rInner = rInner, rOuter = rOuter, zMin = zMin, zMax = zMax)
+
+function RegularPrism(N::Integer, rOuter::R, height::H) where {R<:Real, H<:Real}
+    T = float(promote_type(R,H))
+    RegularPrism( N, T, T(rOuter), T(height/2))
+end
+
+
+@inline in(p::CylindricalPoint, hp::RegularPrism{N, T, <:Real}) where {N,T} = begin
+    _in_z(p, hp.z) && abs(p.r * cos(T(π/N) - mod(p.φ, T(2π/N))) / cos(T(π/N))) <= hp.r
```

Comment:
`abs` not needed

Reply:
We should somehow include a warning if `p.r` is negative, though (probably when creating the `RegularPrism` object). 
Otherwise, this would always be `true`..